File: backend/app/src/models/locations/MainCity.py
from .City import City
from ..buildings.Headquarter import Headquarter
from ..population.Commander import Commander
from ..population.Worker import Worker
# from ..buildings.BuildersHut import BuildersHut
# from ..persons.Builder import Builder


class MainCity(City):
    def __init__(self, name: str, 
                 gold=2000, food=1000, wood=1500, iron=500, 
                 buildings=None,
                 population=None 
                 ) -> None:
        if buildings == None:
            buildings = [Headquarter()]
        if population == None:
            population = [Commander("Gottfried"), Worker("Abrams")]
        super().__init__(name=name, settlement_type="main_city", gold=gold, food=food, wood=wood, iron=iron, buildings=buildings, population=population)
     
    

File: backend_refactoring/CSharpApi/Models/BluePrints/Locations/MainCity.cs

using CSharpApi.Models.BluePrints.Beings;
using CSharpApi.Models.BluePrints.BuildingTypes;

namespace CSharpApi.Models.BluePrints.Locations
{
    public class MainCity : City
    {
        public MainCity(string settlementName, float gold=2000, float food = 2000, float wood = 2000, float iron = 2000,List<BuildingState> buildingStates = null, List<IBuildingList> buildingList = null, List<Population> population = null, string settlementType = "main_city") : base(settlementName, gold, food, wood, iron, buildingStates, population ?? [new Commander("Gottfried"), new Worker("Abrams")], buildingList ?? [new HeadQuarter(), new BuildersHut()], settlementType)
        {
        }
    }
}
